src/controllers/oauth.ts

```diff
@@ -0,0 +1,86 @@
+import axios from 'axios';
+import { Router } from 'express';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const D_CLIENT_ID = process.env.D_CLIENT_ID
+const D_CLIENT_SECRET = process.env.D_CLIENT_SECRET
+const D_REDIRECT_URI = process.env.D_REDIRECT_URI
```

Comment:
validate existence on startup, and error if they do not exist

Reply:
resolved in #12